### PR TITLE
Bug 1906684: handles application group issue with EventSource creation

### PR DESCRIPTION
**Pull request by invincibleJai**
**Fixes**: 
https://issues.redhat.com/browse/ODC-5239

**Analysis / Root cause**: 
Event Source creation fails if user selects no app group and switch to yaml and then to form as application name had text "no application group" as part of annotations

**Solution Description**: 
handles application group issue with EventSource creation

**Unit test coverage report**: 

image

image

**Browser conformance**: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/knative-plugin/src/utils/create-eventsources-utils.ts [hunk 1]
```diff
@@ -466,3 +467,56 @@ export const handleRedirect = (
...
+    application: {
+      ...formDataValues.application,
+      ...(appGroupName &&
+        appGroupName !== formDataValues.application.name && {
+          name: appGroupName,
+          selectedKey: formDataValues.application.selectedKey ? CREATE_APPLICATION_KEY : '',
+        }),
+      ...(!appGroupName && {
+        name: '',
+        selectedKey: '#UNASSIGNED_APP#',
```
Comment (vikram-raj) [inline frontend/packages/knative-plugin/src/utils/create-eventsources-utils.ts:489]: Can we use const `UNASSIGNED_APPLICATIONS_KEY` here?

**Comment by vikram-raj** [comment]
We need to handle the use case for non-admin users. We should display the alert message for the user has only view access for the namespace.

image

**Comment by invincibleJai** [comment]
> We need to handle the use case for non-admin users. We might need to show an alert message instead of loading the form for the user has only view access for the namespace.
> 

@vikram-raj we can get this fixed as separate PR as this is not related to current issue being fixed , WDYT?

**Comment by vikram-raj** [comment]
> > We need to handle the use case for non-admin users. We might need to show an alert message instead of loading the form for the user has only view access for the namespace.
> 
> @vikram-raj we can get this fixed as separate PR/ticket as this is not related to current issue being fixed (can add accesReview even before i.e don't show Kamelet source in catalog if user don't have create access for KB, WDYT?

Agree, we can handle it in a separate issue. And can we hide the Event Source card from the add page if the user has only view access to the namespace?

**Review by vikram-raj — COMMENTED** [review]
Verified it. Works as expected. 

/lgtm

**Pull request merged**